docs/conceptual/core/executor.md

```diff
@@ -2,8 +2,91 @@
 title: Executor
 ---
 
-* Runtime Versioning
+The executor is responsible for dispatching and executing calls into the Substrate runtime.
 
-* Native VS Wasm Execution
+## Runtime Execution
 
-* Forkless Runtime Upgrades
\ No newline at end of file
+The Substrate runtime is compiled into a native executable and a WebAssembly (Wasm) binary.
+
+The native runtime is included as part of the node executable, while the Wasm binary is stored on the blockchain under a well known storage key.
+
+These two representations of the runtime may not be the same. For example: when the runtime is upgraded. The executor determines which version of the runtime to use when dispatching and executing calls.
+
+### Execution Strategy
+
+Substrate allows you to configure an execution strategy during different parts of the blockchain execution process. The strategies are:
+
+- `Native`: Execute with native build (if available, WebAssembly otherwise).
+- `Wasm`: Only execute with the WebAssembly build.
+- `Both`: Execute with both native (where available) and WebAssembly builds.
+- `NativeElseWasm`: Execute with the native build if possible; if it fails, then execute with WebAssembly.
+
+The default execution strategies for the different parts of the blockchain execution process are:
+
+- Syncing: `NativeElseWasm`
+- Block Import: `NativeElseWasm`
+- Block Construction: `Wasm`
+- Off-Chain Worker: `Native`
+- Other: `Native`
+
+### Wasm Execution
+
+The Wasm representation of the Substrate runtime is considered the canonical runtime. Because this Wasm runtime is placed in the blockchain storage, the network must come to consensus about this binary. Thus it can be verified to be consistent across all syncing nodes.
+
+The Wasm execution environment is more restrictive than the native execution environment, because it makes few assumptions about the target machine it will run on. For example, the Wasm runtime always executes in a 32-bit environment.
```

Comment:
It does not makes assumptions about the target machine. Wasm is a vm that dictates the environment, 32bit, max 4gb etc. But you can run the VM everywhere you want.

Reply:
Dumb question, but can I run the Wasm on a machine which does not have at least these specifications? For example a device with only 1 GB of memory.

Reply:
Yes, why not? If you would need 4gb, you would need to give 3gb of swap.
But substrate also does not gives the runtime 4gb of memory, this can be controlled. I think the default should be something around 65mb. But don't quote me on this number.

Reply:
So then this is false:

> The Wasm execution environment can be more restrictive than the native execution environment. For example, the Wasm runtime always executes in a 32-bit environment with a 4 GB memory limit.

I could simply omit any "example" if it is not accurate.